src/engine/GameObjects/MapNode.cxx

```diff
@@ -205,50 +205,74 @@ bool MapNode::isPlacementAllowed(const std::string &newTileID) const
 
   if (tileData)
```

Comment:
Consider moving this line and line 203 to the place where you are using tileData